src/js/components/Pagination/doc.js

```diff
@@ -0,0 +1,98 @@
+import { describe, PropTypes } from 'react-desc';
+
+import { genericProps } from '../../utils/prop-types';
+import { getAvailableAtBadge } from '../../utils/mixins';
+import { themeDocUtils } from '../../utils/themeDocUtils';
+
+export const doc = Pagination => {
+  const DocumentedPagination = describe(Pagination)
+    .availableAt(getAvailableAtBadge('Pagination'))
+    .description(
+      `A control that enables selection of a single page from a 
+      range of pages.`,
+    )
+    .usage(
+      `import { Pagination } from 'grommet';
+      <Pagination />`,
+    )
+    .intrinsicElement('nav');
+
+  DocumentedPagination.propTypes = {
+    ...genericProps,
+    numEdgePages: PropTypes.number
+      .description(
+        `The number of page buttons visible at the start and end of page 
+        range.`,
+      )
+      .defaultValue(1),
+    numItems: PropTypes.number
+      .description('The number of items to paginate.')
+      .defaultValue(undefined),
+    numMiddlePages: PropTypes.number
+      .description(
+        `The number of page buttons visible in the middle of the controls.`,
+      )
+      .defaultValue(3),
```

Comment:
We usually try not to use shortcuts on the API  prop names (i.e. num vs number). Did we try to think about better naming?

Reply:
We did not consider using `number`. I think we were aiming to keep the prop names as consolidated as possible, but open to switching to `numberMiddlePages` to align with grommet conventions. Thoughts here @halocline ?

Reply:
I have changed to use `numberMiddlePages`, etc.